### ISSUE TITLE
📝 Documentation: partial config guidance

### ISSUE THREAD

**Issue by OlivierZal** [JoshuaKGoldberg/eslint-plugin-package-json#1451]
### Documentation Report Checklist

- [x] I have checked the latest `main` branch of the repository.
- [x] I have [searched for related issues](https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

In the documentation, the following works:

```ts
// eslint.config.ts
import packageJson from "eslint-plugin-package-json";

export default [
	// your other ESLint configurations
	packageJson.configs.recommended,
	{
		rules: {
			"package-json/valid-package-definition": "off",
		},
	},
];
```

But the following:

```ts
// eslint.config.ts
import packageJson from "eslint-plugin-package-json";

export default [
	// your other ESLint configurations
	packageJson.configs.recommended,
	{
		rules: {
			"package-json/valid-package-definition": "error", // "error" instead of "off"
		},
	},
];
```

raises the following error:

```
Oops! Something went wrong! :(

ESLint: 9.39.2

A configuration object specifies rule "package-json/valid-package-definition", but could not find plugin "package-json".

Common causes of this problem include:

1. The "package-json" plugin is not defined in your configuration file.
2. The "package-json" plugin is not defined within the same configuration object in which the "package-json/valid-package-definition" rule is applied.
```

To fix it, it should be:
```ts
// eslint.config.ts
import packageJson from "eslint-plugin-package-json";

export default [
	// your other ESLint configurations
	packageJson.configs.recommended,
	{
                plugins: {
                        'package-json': packageJson,
                },
		rules: {
			"package-json/valid-package-definition": "error",
		},
	},
];
```

or

```ts
// eslint.config.ts
import packageJson from "eslint-plugin-package-json";

export default [
	// your other ESLint configurations
	{
                extends: [packageJson.configs.recommended],
                plugins: {
                        'package-json': packageJson,
                },
		rules: {
			"package-json/valid-package-definition": "error",
		},
	},
];
```

**Comment by michaelfaith** [comment]
Good call.  Thanks for pointing this out.  Just one minor tweak to your suggestion: we should probably encourage the use of `defineConfig` from `eslint` as well as using `files`, which is more recently updated guidance from the eslint team.

```ts
// eslint.config.ts
import packageJson from "eslint-plugin-package-json";
import { defineConfig } from "eslint/config";

export default defineConfig(
	// your other ESLint configurations
	{
		extends: [packageJson.configs.recommended],
		files: ["package.json"],
		rules: {
			"package-json/valid-package-definition": "error",
		},
	},
);
```

**Comment by JoshuaKGoldberg** [comment]
Err, that explicit `{ plugins: ... }` should be unnecessary, no? I would think these docs are actually correct as to what the plugin's usage _should_ be - it's the plugin's JS code that's wrong?

**Comment by michaelfaith** [comment]
> it's the plugin's JS code that's wrong?

In what way?  I'm not sure I follow.  Are you referring to the snippet from the original post or my suggested / adjusted version (https://github.com/JoshuaKGoldberg/eslint-plugin-package-json/issues/1451#issuecomment-3662016980)
